### ISSUE TITLE
Scrobbling (Last.fm) sometimes not working

### ISSUE THREAD

**Issue by kroka** [tiliado/nuvolaplayer#189]
Hi,

I have a strange issue with scrobbling of Google Play tracks. Sometimes it's working (Monday, Wednesday), sometimes not at all (Tuesday and today).

Unfortunately even the debug messages do not give a hint on the cause (at least for me). I guess that `AudioScrobblerComponent.vala` should appear somewhere in the logs, however, it does not.

```
$ nuvolaplayer3 --debug --verbose --log-file="nuvola.log"
[Runner:DEBUG      Gtk] Connecting to session manager
[Runner:DEBUG    DioriteGtk] DesktopShell.vala:90: Shell: compiz (null), CSD off
[Runner:DEBUG    DioriteGtk] application.vala:219: XFCE session vanished: org.xfce.SessionManager
[Runner:DEBUG    Nuvola] Connection.vala:132: Network Proxy: system settings
[Runner:DEBUG    Nuvola] WebEngine.vala:66: Nuvola WebKit Extension directory: /usr/lib/nuvolaplayer3
[Runner:WARNING  GLib-GObject] g_object_set_property: construct property "local-storage-directory" for object 'WebKitWebContext' can't be set after construction
[Runner:DEBUG    Nuvola] Notifications.vala:175: Notifications: persistence false, actions false, icons false
[Runner:DEBUG    Nuvola] MPRISProvider.vala:73: Bus acquired: org.mpris.MediaPlayer2.NuvolaAppGooglePlayMusic, registering objects
[Runner:DEBUG    Nuvola] MPRISProvider.vala:90: Bus name acquired: org.mpris.MediaPlayer2.NuvolaAppGooglePlayMusic
[Worker:DEBUG    Nuvola] WebExtension.vala:237: Page 1 created for (null)
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://play.google.com/music/listen#/now, result = true, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] FormatSupport.vala:103: Trying to play /usr/share/nuvolaplayer3/audio/audiotest.mp3.
[Runner:DEBUG    Nuvola] FormatSupport.vala:103: Pipeline state changed from NULL to READY.
[Runner:DEBUG    Nuvola] FormatSupport.vala:103: Pipeline state changed from READY to PAUSED.
[Runner:DEBUG    Nuvola] FormatSupport.vala:103: Pipeline state changed from PAUSED to PLAYING.
[Runner:DEBUG    Nuvola] FormatSupport.vala:103: End of stream for file /usr/share/nuvolaplayer3/audio/audiotest.mp3.
[Runner:DEBUG    Nuvola] FormatSupportCheck.vala:145: Nuvola.WebPlugin: Shockwave Flash (enabled, flash) at /usr/lib/mozilla/plugins/flashplugin-alternative.so: Shockwave Flash 11.2 r202
[Runner:DEBUG    Nuvola] FormatSupportCheck.vala:145: Nuvola.WebPlugin: Java(TM) Plug-in 11.66.2 (enabled, not flash) at /usr/lib/mozilla/plugins/libjavaplugin.so: Next Generation Java Plug-in 11.66.2 for Mozilla browsers
[Runner:CRITICAL  GLib] g_ptr_array_remove_index: assertion 'index_ < rarray->len' failed
[Runner:CRITICAL  GLib] Application will not function properly.
[Worker:DEBUG    Nuvola] WebExtension.vala:140: Window object cleared: https://play.google.com/music/listen#/now, 0x2533100, 0x236d360, 0x7f3c400cf4b0
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] WebAppWindow.vala:170: Sidebar position: 1679
[Runner:CRITICAL  GLib] g_ptr_array_remove_index: assertion 'index_ < rarray->len' failed
[Runner:CRITICAL  GLib] Application will not function properly.
[Runner:INFO     Nuvola] AppRunnerController.vala:236: Logged in as  ()
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://fls.doubleclick.net/activityi;src=2542116;type=andro238;cat=googl515;ord=1;num=7716035572811.961?, result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
https://play-music.gstatic.com/fe/c52f7228062df5870da4245d8e2a3b42/webcomponents-lite.min.js:11:17074: CONSOLE ERROR Refused to get unsafe header "Location"
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://2542116.fls.doubleclick.net/activityi;src=2542116;type=andro238;cat=googl515;ord=1;num=7716035572811.961, result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
https://www.googleadservices.com/pagead/conversion.js:16:111: CONSOLE ERROR Blocked a frame with origin "https://2542116.fls.doubleclick.net" from accessing a frame with origin "https://play.google.com". Protocols, domains, and ports must match.
https://www.googleadservices.com/pagead/conversion.js:16:172: CONSOLE ERROR Blocked a frame with origin "https://2542116.fls.doubleclick.net" from accessing a frame with origin "https://play.google.com". Protocols, domains, and ports must match.
https://www.googleadservices.com/pagead/conversion.js:13:240: CONSOLE ERROR Blocked a frame with origin "https://2542116.fls.doubleclick.net" from accessing a frame with origin "https://play.google.com". Protocols, domains, and ports must match.
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://play.google.com/music/services/streamingloadalltracks?..., result = true, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-pause
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-stop
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-play
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-go-previous
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-go-next
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://0.client-channel.google.com/client-channel/client?..., result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://clients4.google.com/invalidation/lcs/client?..., result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://accounts.google.com/o/oauth2/postmessageRelay?parent=https%3A%2F%2Fplay.google.com#rpctoken=864510501&forcesecure=1, result = true, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
Vector smash protection is enabled.
Failed to open VDPAU backend libvdpau_va_gl.so: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden
Failed to open VDPAU backend libvdpau_va_gl.so: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-play
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] WebEngine.vala:464: Navigation, current window: uri = https://plus.google.com/u/0/_/notifications/frame?..., result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
 	[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-pause
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-stop
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-play
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
...

$ cat nuvola.log 
[Master:DEBUG      Gtk] Connecting to session manager
[Master:DEBUG    DioriteGtk] DesktopShell.vala:90: Shell: compiz (null), CSD off
...
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Play Music at /usr/share/nuvolaplayer3/web_apps/google_play_music, version 5.8
...
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Test at /usr/share/nuvolaplayer3/web_apps/test, version 1.0
[Master:WARNING  Nuvola] WebAppRegistry.vala:165: Unable to load web app from /usr/share/nuvolaplayer3/web_apps/template: /usr/share/nuvolaplayer3/web_apps/template/metadata.json is not a file
[Master:DEBUG    DioriteGtk] application.vala:219: XFCE session vanished: org.xfce.SessionManager
...
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Play Music at /usr/share/nuvolaplayer3/web_apps/google_play_music, version 5.8
....
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Test at /usr/share/nuvolaplayer3/web_apps/test, version 1.0
[Master:WARNING  Nuvola] WebAppRegistry.vala:165: Unable to load web app from /usr/share/nuvolaplayer3/web_apps/template: /usr/share/nuvolaplayer3/web_apps/template/metadata.json is not a file
[Master:INFO     Nuvola] WebAppRegistry.vala:89: Using web app Google Play Music, version 5.8, data dir /usr/share/nuvolaplayer3/web_apps/google_play_music
[Master:DEBUG    Nuvola] MasterController.vala:450: Launch app runner for 'google_play_music': /usr/lib/nuvolaplayer3/apprunner -D -a /usr/share/nuvolaplayer3/web_apps/google_play_music 
[Master:DEBUG    Nuvola] MasterController.vala:274: Connected to runner server for 'google_play_music'.
[Master:DEBUG    Nuvola] MediaKeys.vala:99: GNOME settings daemon appeared: org.gnome.SettingsDaemon, :1.18
```

* Nuvola Player 3 Beta 3.0.0+201512061359.818b587
  * Revision 818b587ae0c6990044b88947edc30cfc6389c21e
* Ubuntu 15.10

Of course, scrobbling is enabled and I am connected to my Last.fm account in the settings.

(sorry, I've published this issue by accident when it was not finished ;-))

**Comment by fenryxo** [comment]
Hello @kroka. Thanks for taking time to report this issue. It may be related to tiliado/nuvola-app-google-play-music#6. Could you check whether the issue still persist in a freshly released Google Play Music script 5.9? Thanks!

**Comment by kroka** [comment]
Indeed it does work again with version 5.9. Thanks for the update!
